diff --git a/commands/kpMacroCommand.cpp b/commands/kpMacroCommand.cpp
--- a/commands/kpMacroCommand.cpp
+++ b/commands/kpMacroCommand.cpp
@@ -99,7 +99,7 @@ void kpMacroCommand::execute ()
 
     for (QLinkedList <kpCommand *>::const_iterator it = m_commandList.begin ();
          it != m_commandList.end ();
-         it++)
+         ++it)
     {
     #if DEBUG_KP_COMMAND_HISTORY
         kDebug () << "\texecuting " << (*it)->name ();
diff --git a/document/kpDocumentSaveOptions.cpp b/document/kpDocumentSaveOptions.cpp
--- a/document/kpDocumentSaveOptions.cpp
+++ b/document/kpDocumentSaveOptions.cpp
@@ -516,7 +516,7 @@ int kpDocumentSaveOptions::mimeTypeMaximumColorDepth (const QString &mimeType)
     const QString mimeTypeColon = mimeType + QLatin1String (":");
     for (QStringList::const_iterator it = mimeTypeList.begin ();
          it != mimeTypeList.end ();
-         it++)
+         ++it)
     {
         if ((*it).startsWith (mimeTypeColon))
         {
diff --git a/document/kpDocument_Save.cpp b/document/kpDocument_Save.cpp
--- a/document/kpDocument_Save.cpp
+++ b/document/kpDocument_Save.cpp
@@ -252,7 +252,7 @@ bool kpDocument::savePixmapToDevice (const QImage &image,
     QList <QString> keyList = metaInfo.textKeys ();
     for (QList <QString>::const_iterator it = keyList.constBegin ();
          it != keyList.constEnd ();
-         it++)
+         ++it)
     {
         imageToSave.setText (*it, metaInfo.text (*it));
     }
diff --git a/imagelib/kpDocumentMetaInfo.cpp b/imagelib/kpDocumentMetaInfo.cpp
--- a/imagelib/kpDocumentMetaInfo.cpp
+++ b/imagelib/kpDocumentMetaInfo.cpp
@@ -150,7 +150,7 @@ void kpDocumentMetaInfo::printDebug (const QString &prefix) const
     QList <QString> keyList = textKeys ();
     for (QList <QString>::const_iterator it = keyList.constBegin ();
          it != keyList.constEnd ();
-         it++)
+         ++it)
     {
         kDebug () << "key=" << (*it)
                   << " text=" << text (*it)
diff --git a/imagelib/kpFloodFill.cpp b/imagelib/kpFloodFill.cpp
--- a/imagelib/kpFloodFill.cpp
+++ b/imagelib/kpFloodFill.cpp
@@ -340,7 +340,7 @@ void kpFloodFill::prepare ()
 
     for (QLinkedList <kpFillLine>::ConstIterator it = d->fillLines.begin ();
          it != d->fillLines.end ();
-         it++)
+         ++it)
     {
     #if DEBUG_KP_FLOOD_FILL && 0
         kDebug () << "Expanding from y=" << (*it).m_y
diff --git a/imagelib/kpPainter.cpp b/imagelib/kpPainter.cpp
--- a/imagelib/kpPainter.cpp
+++ b/imagelib/kpPainter.cpp
@@ -435,7 +435,7 @@ static QRect WashLineHelper (QPainter *rgbPainter, void *data)
     QList <QPoint> points = kpPainter::interpolatePoints (pack->startPoint, pack->endPoint);
     for (QList <QPoint>::const_iterator pit = points.constBegin ();
             pit != points.constEnd ();
-            pit++)
+            ++pit)
     {
         // OPT: This may be reading and possibly writing pixels that were
         //      visited on a previous iteration, since the pen is usually
diff --git a/layers/selections/text/kpTextSelection.cpp b/layers/selections/text/kpTextSelection.cpp
--- a/layers/selections/text/kpTextSelection.cpp
+++ b/layers/selections/text/kpTextSelection.cpp
@@ -302,7 +302,7 @@ QString kpTextSelection::TextForTextLines (const QList <QString> &textLines)
 
     for (QList <QString>::const_iterator it = textLines.begin () + 1;
          it != textLines.end ();
-         it++)
+         ++it)
     {
         bigString += QLatin1String ("\n");
         bigString += (*it);
diff --git a/mainWindow/kpMainWindow_Edit.cpp b/mainWindow/kpMainWindow_Edit.cpp
--- a/mainWindow/kpMainWindow_Edit.cpp
+++ b/mainWindow/kpMainWindow_Edit.cpp
@@ -500,7 +500,7 @@ void kpMainWindow::pasteText (const QString &text,
         int width = 0;
         for (QList <QString>::const_iterator it = textLines.constBegin ();
              it != textLines.constEnd ();
-             it++)
+             ++it)
         {
             const int w = fontMetrics.width (*it);
             if (w > width)
diff --git a/mainWindow/kpMainWindow_File.cpp b/mainWindow/kpMainWindow_File.cpp
--- a/mainWindow/kpMainWindow_File.cpp
+++ b/mainWindow/kpMainWindow_File.cpp
@@ -465,7 +465,7 @@ void kpMainWindow::slotOpen ()
 
     for (KUrl::List::const_iterator it = urls.begin ();
          it != urls.end ();
-         it++)
+         ++it)
     {
         open (*it);
     }
diff --git a/mainWindow/kpMainWindow_Tools.cpp b/mainWindow/kpMainWindow_Tools.cpp
--- a/mainWindow/kpMainWindow_Tools.cpp
+++ b/mainWindow/kpMainWindow_Tools.cpp
@@ -195,7 +195,7 @@ void kpMainWindow::createToolBox ()
 
     for (QList <kpTool *>::const_iterator it = d->tools.constBegin ();
          it != d->tools.constEnd ();
-         it++)
+         ++it)
     {
         d->toolToolBar->registerTool (*it);
     }
@@ -244,7 +244,7 @@ void kpMainWindow::enableToolsDocumentActions (bool enable)
 
     for (QList <kpTool *>::const_iterator it = d->tools.constBegin ();
          it != d->tools.constEnd ();
-         it++)
+         ++it)
     {
         kpToolAction *action = (*it)->action ();
         if (action)
@@ -527,7 +527,7 @@ int kpMainWindow::toolNumber () const
     int number = 0;
     for (QList <kpTool *>::const_iterator it = d->tools.constBegin ();
          it != d->tools.constEnd ();
-         it++)
+         ++it)
     {
         if (*it == tool ())
             return number;
diff --git a/mainWindow/kpMainWindow_View_Zoom.cpp b/mainWindow/kpMainWindow_View_Zoom.cpp
--- a/mainWindow/kpMainWindow_View_Zoom.cpp
+++ b/mainWindow/kpMainWindow_View_Zoom.cpp
@@ -158,7 +158,7 @@ void kpMainWindow::sendZoomListToActionZoom ()
     const QList <int>::ConstIterator zoomListEnd (d->zoomList.end ());
     for (QList <int>::ConstIterator it = d->zoomList.constBegin ();
          it != zoomListEnd;
-         it++)
+         ++it)
     {
         items << ::ZoomLevelToString (*it);
     }
diff --git a/tools/flow/kpToolFlowPixmapBase.cpp b/tools/flow/kpToolFlowPixmapBase.cpp
--- a/tools/flow/kpToolFlowPixmapBase.cpp
+++ b/tools/flow/kpToolFlowPixmapBase.cpp
@@ -59,7 +59,7 @@ QRect kpToolFlowPixmapBase::drawLine (const QPoint &thisPoint, const QPoint &las
 
     for (QList <QPoint>::const_iterator pit = points.constBegin ();
          pit != points.constEnd ();
-         pit++)
+         ++pit)
     {
         const QPoint point =
             hotRectForMousePointAndBrushWidthHeight (
diff --git a/views/manager/kpViewManager.cpp b/views/manager/kpViewManager.cpp
--- a/views/manager/kpViewManager.cpp
+++ b/views/manager/kpViewManager.cpp
@@ -152,7 +152,7 @@ kpView *kpViewManager::viewUnderCursor (bool usingQt) const
     {
         for (QLinkedList <kpView *>::const_iterator it = d->views.begin ();
              it != d->views.end ();
-             it++)
+             ++it)
         {
             if ((*it)->underMouse ())
                 return (*it);
@@ -211,7 +211,7 @@ bool kpViewManager::hasAViewWithFocus () const
 {
     for (QLinkedList <kpView *>::const_iterator it = d->views.begin ();
          it != d->views.end ();
-         it++)
+         ++it)
     {
         if ((*it)->isActiveWindow ())
             return true;
@@ -227,7 +227,7 @@ void kpViewManager::setCursor (const QCursor &cursor)
 {
     for (QLinkedList <kpView *>::const_iterator it = d->views.begin ();
          it != d->views.end ();
-         it++)
+         ++it)
     {
         (*it)->setCursor (cursor);
     }
@@ -242,7 +242,7 @@ void kpViewManager::unsetCursor ()
 {
     for (QLinkedList <kpView *>::const_iterator it = d->views.begin ();
          it != d->views.end ();
-         it++)
+         ++it)
     {
         (*it)->unsetCursor ();
     }
diff --git a/views/manager/kpViewManager_ViewUpdates.cpp b/views/manager/kpViewManager_ViewUpdates.cpp
--- a/views/manager/kpViewManager_ViewUpdates.cpp
+++ b/views/manager/kpViewManager_ViewUpdates.cpp
@@ -77,7 +77,7 @@ void kpViewManager::restoreQueueUpdates ()
     {
         for (QLinkedList <kpView *>::const_iterator it = d->views.begin ();
              it != d->views.end ();
-             it++)
+             ++it)
         {
             (*it)->updateQueuedArea ();
         }
@@ -204,7 +204,7 @@ void kpViewManager::updateViews (const QRect &docRect)
 
     for (QLinkedList <kpView *>::const_iterator it = d->views.begin ();
          it != d->views.end ();
-         it++)
+         ++it)
     {
         kpView *view = *it;
 
@@ -255,7 +255,7 @@ void kpViewManager::adjustViewsToEnvironment ()
 #endif
     for (QLinkedList <kpView *>::const_iterator it = d->views.begin ();
          it != d->views.end ();
-         it++)
+         ++it)
     {
         kpView *view = *it;
 
diff --git a/widgets/toolbars/kpToolToolBar.cpp b/widgets/toolbars/kpToolToolBar.cpp
--- a/widgets/toolbars/kpToolToolBar.cpp
+++ b/widgets/toolbars/kpToolToolBar.cpp
@@ -106,7 +106,7 @@ kpToolToolBar::kpToolToolBar(const QString &label, int colsOrRows, QWidget *pare
 
     for (QList <kpToolWidgetBase *>::const_iterator it = m_toolWidgets.constBegin ();
          it != m_toolWidgets.constEnd ();
-         it++)
+         ++it)
     {
         connect (*it, SIGNAL (optionSelected (int, int)),
                  this, SIGNAL (toolWidgetOptionSelected ()));
@@ -141,7 +141,7 @@ void kpToolToolBar::registerTool (kpTool *tool)
 {
     for (QList <kpButtonToolPair>::const_iterator it = m_buttonToolPairs.constBegin ();
          it != m_buttonToolPairs.constEnd ();
-         it++)
+         ++it)
     {
         if ((*it).m_tool == tool)
             return;
@@ -178,7 +178,7 @@ void kpToolToolBar::unregisterTool (kpTool *tool)
 {
     for (QList <kpButtonToolPair>::iterator it = m_buttonToolPairs.begin ();
          it != m_buttonToolPairs.end ();
-         it++)
+         ++it)
     {
         if ((*it).m_tool == tool)
         {
@@ -201,7 +201,7 @@ void kpToolToolBar::unregisterAllTools ()
 {
     for (QList <kpButtonToolPair>::iterator it = m_buttonToolPairs.begin ();
          it != m_buttonToolPairs.end ();
-         it++)
+         ++it)
     {
        delete ((*it).m_button);
     }
@@ -235,7 +235,7 @@ void kpToolToolBar::selectTool (const kpTool *tool, bool reselectIfSameTool)
     {
         for (QList <kpButtonToolPair>::iterator it = m_buttonToolPairs.begin ();
             it != m_buttonToolPairs.end ();
-            it++)
+            ++it)
         {
             if ((*it).m_tool == tool)
             {
@@ -293,7 +293,7 @@ void kpToolToolBar::hideAllToolWidgets ()
 {
     for (QList <kpToolWidgetBase *>::const_iterator it = m_toolWidgets.constBegin ();
          it != m_toolWidgets.constEnd ();
-         it++)
+         ++it)
     {
         (*it)->hide ();
     }
@@ -308,7 +308,7 @@ kpToolWidgetBase *kpToolToolBar::shownToolWidget (int which) const
 
     for (QList <kpToolWidgetBase *>::const_iterator it = m_toolWidgets.constBegin ();
          it != m_toolWidgets.constEnd ();
-         it++)
+         ++it)
     {
         if (!(*it)->isHidden ())
         {
@@ -336,7 +336,7 @@ void kpToolToolBar::slotToolButtonClicked ()
     kpTool *tool = 0;
     for (QList <kpButtonToolPair>::iterator it = m_buttonToolPairs.begin ();
          it != m_buttonToolPairs.end ();
-         it++)
+         ++it)
     {
         if ((*it).m_button == b)
         {
@@ -425,7 +425,7 @@ void kpToolToolBar::slotToolActionToolTipChanged ()
 
     for (QList <kpButtonToolPair>::const_iterator it = m_buttonToolPairs.constBegin ();
         it != m_buttonToolPairs.constEnd ();
-        it++)
+        ++it)
     {
         if (tool == (*it).m_tool)
         {
@@ -469,7 +469,7 @@ void kpToolToolBar::adjustToOrientation(Qt::Orientation o)
 
     for (QList <kpButtonToolPair>::iterator it = m_buttonToolPairs.begin ();
          it != m_buttonToolPairs.end ();
-         it++)
+         ++it)
     {
         addButton ((*it).m_button, o, num);
         num++;
@@ -477,7 +477,7 @@ void kpToolToolBar::adjustToOrientation(Qt::Orientation o)
 
     for (QList <kpToolWidgetBase *>::const_iterator it = m_toolWidgets.constBegin ();
          it != m_toolWidgets.constEnd ();
-         it++)
+         ++it)
     {
         if (*it)
         {
@@ -525,7 +525,7 @@ void kpToolToolBar::slotIconSizeChanged(const QSize &size)
 {
     for (QList <kpButtonToolPair>::iterator it = m_buttonToolPairs.begin ();
          it != m_buttonToolPairs.end ();
-         it++)
+         ++it)
     {
         (*it).m_button->setIconSize(size);
     }
