Index: kptool.cpp
===================================================================
--- kptool.cpp	(revision 530107)
+++ kptool.cpp	(revision 533650)
@@ -153,7 +153,7 @@ void kpTool::createAction ()
     m_action = new kpToolAction (text (), iconName (), shortcutForKey (m_key),
                                  this, SLOT (slotActionActivated ()),
                                  m_mainWindow->actionCollection (), name ());
-    m_action->setExclusiveGroup (QLatin1String ("Tool Box Actions"));
+    //m_action->setExclusiveGroup (QLatin1String ("Tool Box Actions"));
     m_action->setWhatsThis (description ());
 
     connect (m_action, SIGNAL (toolTipChanged (const QString &)),
@@ -192,11 +192,13 @@ QString kpTool::toolTipForTextAndShortcu
 {
     for (int i = 0; i < (int) shortcut.count (); i++)
     {
+#if 0
         const KKeySequence seq = shortcut.seq (i);
         if (seq.count () == 1 && containsSingleKeyTrigger (seq))
+#endif
         {
             return i18nc ("<Tool Name> (<Single Accel Key>)",
-                          "%1 (%2)", text, seq.toString ());
+                          "%1 (%2)", text, shortcut.toString ());
         }
     }
 
@@ -235,10 +237,10 @@ KShortcut kpTool::shortcutForKey (int ke
 
     if (key)
     {
-        shortcut.append (KKeySequence (KKey (key)));
+        shortcut.append (KShortcut (key));
         // (CTRL+<key>, ALT+<key>, CTRL+ALT+<key>, CTRL+SHIFT+<key>
         //  all clash with global KDE shortcuts)
-        shortcut.append (KKeySequence (KKey (Qt::ALT + Qt::SHIFT + key)));
+        shortcut.append (KShortcut (Qt::ALT + Qt::SHIFT + key));
     }
 
     return shortcut;
@@ -259,19 +261,6 @@ bool kpTool::keyIsText (int key)
 }
 
 // public static
-bool kpTool::containsSingleKeyTrigger (const KKeySequence &seq)
-{
-    for (int i = 0; i < (int) seq.count (); i++)
-    {
-        const KKey key = seq.key (i);
-        if (keyIsText (key.keyCodeQt ()))
-            return true;
-    }
-
-    return false;
-}
-
-// public static
 bool kpTool::containsSingleKeyTrigger (const KShortcut &shortcut,
     KShortcut *shortcutWithoutSingleKeyTriggers)
 {
@@ -282,6 +271,7 @@ bool kpTool::containsSingleKeyTrigger (c
     KShortcut newShortcut;
     bool needNewShortcut = false;
 
+#if 0
     for (int i = 0; i < (int) shortcut.count (); i++)
     {
         const KKeySequence seq = shortcut.seq (i);
@@ -295,10 +285,11 @@ bool kpTool::containsSingleKeyTrigger (c
             newShortcut.append (seq);
         }
     }
+#endif
 
 
     if (needNewShortcut && shortcutWithoutSingleKeyTriggers)
-        *shortcutWithoutSingleKeyTriggers = newShortcut;
+        *shortcutWithoutSingleKeyTriggers = shortcut;
 
     return needNewShortcut;
 }
Index: tools/kptoolaction.cpp
===================================================================
--- tools/kptoolaction.cpp	(revision 530107)
+++ tools/kptoolaction.cpp	(revision 533650)
@@ -75,9 +75,11 @@ void kpToolAction::setText (const QStrin
 // public slot virtual [base KAction]
 bool kpToolAction::setShortcut (const KShortcut &shortcut)
 {
-    bool ret = KToggleAction::setShortcut (shortcut);
+#warning "kde4: port it"
+    //bool ret = KToggleAction::setShortcut (shortcut);
     updateToolTip ();
-    return ret;
+    //return ret;
+	return false;
 }
 
 
Index: kpmainwindow_settings.cpp
===================================================================
--- kpmainwindow_settings.cpp	(revision 530107)
+++ kpmainwindow_settings.cpp	(revision 533650)
@@ -136,7 +136,7 @@ void kpMainWindow::slotKeyBindings ()
         enableActionsSingleKeyTriggers (true);
 
 
-    if (KKeyDialog::configure (actionCollection (), this))
+    if (KKeyDialog::configure (actionCollection (), KKeyChooser::LetterShortcutsAllowed, this))
     {
     #if DEBUG_KP_MAIN_WINDOW
         kDebug () << "\tdialog accepted" << endl;
Index: kptool.h
===================================================================
--- kptool.h	(revision 530107)
+++ kptool.h	(revision 533650)
@@ -48,7 +48,6 @@ class QMouseEvent;
 class QPixmap;
 class QWheelEvent;
 
-class KKeySequence;
 class KShortcut;
 
 class kpColor;
@@ -112,7 +111,6 @@ public:
     KShortcut shortcut () const;
 
     static bool keyIsText (int key);
-    static bool containsSingleKeyTrigger (const KKeySequence &seq);
     static bool containsSingleKeyTrigger (const KShortcut &shortcut,
         KShortcut *shortcutWithoutSingleKeyTriggers);
 
Index: kpmainwindow_help.cpp
===================================================================
--- kpmainwindow_help.cpp	(revision 530107)
+++ kpmainwindow_help.cpp	(revision 533650)
@@ -89,7 +89,7 @@ static QString printScreenShortcutString
     else
     {
         // (localised)
-        return KKey (Qt::CTRL + Qt::Key_Print).toString ();
+        return KShortcut (Qt::CTRL + Qt::Key_Print).toString ();
     }
 }
 
