Index: tools/kptoolresizescale.cpp
===================================================================
--- tools/kptoolresizescale.cpp	(revision 535888)
+++ tools/kptoolresizescale.cpp	(revision 536486)
@@ -695,10 +695,9 @@ void kpToolResizeScaleDialog::createOper
     resizeScaleButtonGroup->insert (m_smoothScaleButton);
 
 
-    QGridLayout *operationLayout = new QGridLayout (m_operationGroupBox,
-                                                    1, 2,
-                                                    marginHint () * 2/*don't overlap groupbox title*/,
-                                                    spacingHint ());
+    QGridLayout *operationLayout = new QGridLayout (m_operationGroupBox );
+    operationLayout->setMargin( marginHint () * 2/*don't overlap groupbox title*/ );
+    operationLayout->setSpacing( spacingHint ());
 
     operationLayout->addWidget (m_resizeButton, 0, 0, Qt::AlignCenter);
     //operationLayout->addWidget (m_resizeLabel, 1, 0, Qt::AlignCenter);
@@ -770,8 +769,9 @@ void kpToolResizeScaleDialog::createDime
     percentLabel->setBuddy (m_percentWidthInput);
 
 
-    QGridLayout *dimensionsLayout = new QGridLayout (m_dimensionsGroupBox,
-                                                     5, 4, marginHint () * 2, spacingHint ());
+    QGridLayout *dimensionsLayout = new QGridLayout (m_dimensionsGroupBox );
+    dimensionsLayout->setMargin( marginHint () * 2 );
+    dimensionsLayout->setSpacing( spacingHint ());
     dimensionsLayout->setColStretch (1/*column*/, 1);
     dimensionsLayout->setColStretch (3/*column*/, 1);
 
Index: tools/kptoolpen.cpp
===================================================================
--- tools/kptoolpen.cpp	(revision 535888)
+++ tools/kptoolpen.cpp	(revision 536486)
@@ -241,7 +241,7 @@ void kpToolPen::hover (const QPoint &poi
     {
         m_mouseButton = 0;
 
-        kpTempPixmap::RenderMode renderMode;
+        kpTempPixmap::RenderMode renderMode = kpTempPixmap::SetPixmap; // default
         QPixmap cursorPixmapForTempPixmap = m_cursorPixmap;
 
         if (m_mode & SquareBrushes)
Index: tools/kptoolrotate.cpp
===================================================================
--- tools/kptoolrotate.cpp	(revision 535888)
+++ tools/kptoolrotate.cpp	(revision 536486)
@@ -32,7 +32,6 @@
 
 #include <qapplication.h>
 #include <qbuttongroup.h>
-#include <q3gridlayout.h>
 #include <qgroupbox.h>
 #include <qlabel.h>
 #include <qlayout.h>
@@ -285,8 +284,9 @@ void kpToolRotateDialog::createDirection
     m_clockwiseRadioButton->setChecked (s_lastIsClockwise);
 
 
-    QGridLayout *directionLayout = new QGridLayout (directionGroupBox,
-                                                    2, 2, marginHint () * 2, spacingHint ());
+    QGridLayout *directionLayout = new QGridLayout (directionGroupBox );
+    directionLayout->setSpacing( spacingHint() );
+    directionLayout->setMargin( marginHint () * 2 );
     directionLayout->addWidget (antiClockwisePixmapLabel, 0, 0, Qt::AlignCenter);
     directionLayout->addWidget (clockwisePixmapLabel, 0, 1, Qt::AlignCenter);
     directionLayout->addWidget (m_antiClockwiseRadioButton, 1, 0, Qt::AlignCenter);
@@ -320,9 +320,9 @@ void kpToolRotateDialog::createAngleGrou
     m_angleCustomRadioButton->setChecked (true);
 
 
-    QGridLayout *angleLayout = new QGridLayout (angleGroupBox,
-                                                6, 3,
-                                                marginHint () * 2, spacingHint ());
+    QGridLayout *angleLayout = new QGridLayout (angleGroupBox );
+    angleLayout->setMargin( marginHint () * 2 );
+    angleLayout->setSpacing( spacingHint ());
 
     angleLayout->addMultiCellWidget (m_angle90RadioButton, 0, 0, 0, 2);
     angleLayout->addMultiCellWidget (m_angle180RadioButton, 1, 1, 0, 2);
Index: tools/kptoolpreviewdialog.cpp
===================================================================
--- tools/kptoolpreviewdialog.cpp	(revision 535888)
+++ tools/kptoolpreviewdialog.cpp	(revision 536486)
@@ -92,8 +92,8 @@ kpToolPreviewDialog::kpToolPreviewDialog
         createPreviewGroupBox ();
 
 
-    m_gridLayout = new QGridLayout (baseWidget, 4, 2,
-                                    0/*margin*/, spacingHint ());
+    m_gridLayout = new QGridLayout (baseWidget );
+    m_gridLayout->setSpacing( spacingHint() );
     m_gridNumRows = reserveTopRow ? 1 : 0;
     if (m_dimensionsGroupBox || m_previewGroupBox)
     {
@@ -150,9 +150,9 @@ void kpToolPreviewDialog::createDimensio
     m_afterTransformDimensionsLabel = new QLabel (m_dimensionsGroupBox);
 
 
-    QGridLayout *dimensionsLayout = new QGridLayout (m_dimensionsGroupBox,
-                                                     2, 2,
-                                                     marginHint () * 2, spacingHint ());
+    QGridLayout *dimensionsLayout = new QGridLayout (m_dimensionsGroupBox );
+    dimensionsLayout->setMargin( marginHint () * 2 );
+    dimensionsLayout->setSpacing( spacingHint ());
 
     dimensionsLayout->addWidget (originalLabel, 0, 0, Qt::AlignBottom);
     dimensionsLayout->addWidget (originalDimensionsLabel, 0, 1, Qt::AlignBottom);
Index: widgets/kpcolorsimilaritycube.cpp
===================================================================
--- widgets/kpcolorsimilaritycube.cpp	(revision 535888)
+++ widgets/kpcolorsimilaritycube.cpp	(revision 536486)
@@ -248,8 +248,8 @@ void kpColorSimilarityCube::drawFace (QP
     #if DEBUG_KP_COLOR_SIMILARITY_CUBE
         kDebug () << "\tnot enabled - making us grey" << endl;
     #endif
-        colors [0] = colorGroup ().background ();
-        colors [1] = colorGroup ().background ();
+        colors [0] = palette ().color( QPalette::Background );
+        colors [1] = palette ().color( QPalette::Background );
     }
 
 #if DEBUG_KP_COLOR_SIMILARITY_CUBE
@@ -275,7 +275,7 @@ void kpColorSimilarityCube::drawContents
     QRect cr (contentsRect ());
 
     QPixmap backBuffer (cr.width (), cr.height ());
-    backBuffer.fill (colorGroup ().background ());
+    backBuffer.fill ( palette().color( QPalette::Background) );
 
     QPainter backBufferPainter (&backBuffer);
 
@@ -591,24 +591,19 @@ void kpToolToolBar::setOrientation (Qt::
         m_baseLayout = new QBoxLayout (m_baseWidget, QBoxLayout::TopToBottom,
                                        5/*margin*/,
                                        10/*spacing*/);
-        m_toolLayout = new QGridLayout (m_baseLayout,
-                                        5/*arbitrary rows since toolBar auto-expands*/,
-                                        m_vertCols,
-                                        0/*margin*/,
-                                        0/*spacing*/);
     }
     else // if (o == Qt::Horizontal)
     {
         m_baseLayout = new QBoxLayout (m_baseWidget, QBoxLayout::LeftToRight,
                                        5/*margin*/,
                                        10/*spacing*/);
-        m_toolLayout = new QGridLayout (m_baseLayout,
-                                        m_vertCols/*rows in this case, since horiz*/,
-                                        5/*arbitrary cols since toolBar auto-expands*/,
-                                        0/*margin*/,
-                                        0/*spacing*/);
     }
 
+    m_toolLayout = new QGridLayout ();
+    m_baseLayout->addItem( m_toolLayout );
+    m_toolLayout->setMargin( 0 );
+    m_toolLayout->setSpacing( 0 );
+
     int num = 0;
 
     for (QList <kpButtonToolPair>::iterator it = m_buttonToolPairs.begin ();
Index: widgets/kpcolortoolbar.cpp
===================================================================
--- widgets/kpcolortoolbar.cpp	(revision 535888)
+++ widgets/kpcolortoolbar.cpp	(revision 536486)
@@ -318,14 +318,14 @@ void kpDualColorButton::drawContents (QP
     else
     {
         backBufferPainter.fillRect (m_backBuffer->rect (),
-                                    colorGroup ().color (QColorGroup::Background));
+                                    palette ().color (QPalette::Background));
     }
 
     QPixmap swapPixmap = UserIcon ("colorbutton_swap_16x16");
     if (!isEnabled ())
     {
         // swapPixmap has a mask after all
-        swapPixmap.fill (colorGroup ().color (QColorGroup::Dark));
+        swapPixmap.fill (palette().color (QPalette::Dark));
     }
     backBufferPainter.drawPixmap (swapPixmapRect ().topLeft (), swapPixmap);
 
@@ -346,8 +346,8 @@ void kpDualColorButton::drawContents (QP
             backBufferPainter.drawPixmap (bgRectInside, UserIcon ("color_transparent_26x26"));
     }
     else
-        backBufferPainter.fillRect (bgRectInside, colorGroup ().color (QColorGroup::Button));
-    qDrawShadePanel (&backBufferPainter, bgRect, colorGroup (),
+        backBufferPainter.fillRect (bgRectInside, palette().color (QPalette::Button));
+    qDrawShadePanel (&backBufferPainter, bgRect, palette(),
                      false/*not sunken*/, 2/*lineWidth*/,
                      0/*never fill*/);
 
@@ -366,8 +366,8 @@ void kpDualColorButton::drawContents (QP
             backBufferPainter.drawPixmap (fgRectInside, UserIcon ("color_transparent_26x26"));
     }
     else
-        backBufferPainter.fillRect (fgRectInside, colorGroup ().color (QColorGroup::Button));
-    qDrawShadePanel (&backBufferPainter, fgRect, colorGroup (),
+        backBufferPainter.fillRect (fgRectInside, palette ().color (QPalette::Button));
+    qDrawShadePanel (&backBufferPainter, fgRect, palette (),
                      false/*not sunken*/, 2/*lineWidth*/,
                      0/*never fill*/);
 
Index: Messages.sh
===================================================================
--- Messages.sh	(revision 0)
+++ Messages.sh	(revision 536486)
@@ -0,0 +1,3 @@
+#! /bin/sh
+$EXTRACTRC `find . -name "*.rc" -o -name "*.ui" ` >> rc.cpp
+$XGETTEXT `find . -name "*.cpp" -o -name "*.h"` -o $podir/kolourpaint.pot
Index: kpdocumentsaveoptionswidget.cpp
===================================================================
--- kpdocumentsaveoptionswidget.cpp	(revision 535888)
+++ kpdocumentsaveoptionswidget.cpp	(revision 536486)
@@ -85,8 +85,9 @@ KDialogBase (parent, name, false/*non-mo
     //setMainWidget (baseWidget);
 
 
-    QGridLayout *lay = new QGridLayout (baseWidget, 2, 1,
-                                        KDialog::marginHint (), KDialog::spacingHint ());
+    QGridLayout *lay = new QGridLayout ( baseWidget );
+    lay->setMargin( KDialog::marginHint () );
+    lay->setSpacing( KDialog::spacingHint ());
 
     m_filePixmapLabel = new kpResizeSignallingLabel (baseWidget);
     m_fileSizeLabel = new QLabel (baseWidget);
Index: kpviewscrollablecontainer.cpp
===================================================================
--- kpviewscrollablecontainer.cpp	(revision 535888)
+++ kpviewscrollablecontainer.cpp	(revision 536486)
@@ -188,7 +188,7 @@ void kpGrip::updatePixmap ()
         return;
 
     QPixmap pixmap (width (), height ());
-    pixmap.fill (colorGroup ().highlight ());
+    pixmap.fill (palette().color( QPalette::Highlight ) );
     kpPixmapFX::ensureTransparentAt (&pixmap, pixmap.rect ());
     const QRect hr = hotRect ();
 #if DEBUG_KP_VIEW_SCROLLABLE_CONTAINER
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 535888)
+++ Makefile.am	(revision 536486)
@@ -61,19 +61,4 @@ rc_DATA = kolourpaintui.rc
 
 xdg_apps_DATA = kolourpaint.desktop
 
-messages: rc.cpp
-	$(EXTRACTRC) *.rc *.ui \
-		compat/*.rc compat/*.ui \
-		cursors/*.rc cursors/*.ui \
-		pixmapfx/*.rc pixmapfx/*.ui \
-		tools/*.rc tools/*.ui \
-		widgets/*.rc widgets/*.ui \
-		>> rc.cpp
-	$(XGETTEXT) *.cpp *.h \
-		compat/*.cpp compat/*.h \
-		cursors/*.cpp cursors/*.h \
-		pixmapfx/*.cpp pixmapfx/*.h \
-		tools/*.cpp tools/*.h \
-		widgets/*.cpp widgets/*.h \
-		-o $(podir)/kolourpaint.pot
 
