OK

Index: kpdocument.cpp
===================================================================
--- kpdocument.cpp	(revision 543519)
+++ kpdocument.cpp	(revision 543520)
@@ -414,7 +414,7 @@ bool kpDocument::save (bool overwritePro
             i18n ("Could not save image - insufficient information."),
             i18n ("URL: %1\n"
                   "Mimetype: %2",
-                  prettyURL (),
+                  prettyUrl (),
                   m_saveOptions->mimeType ().isEmpty () ?
                           i18n ("<empty>") :
                           m_saveOptions->mimeType ()),
@@ -771,10 +771,10 @@ QString kpDocument::prettyURLForURL (con
     if (url.isEmpty ())
         return i18n ("Untitled");
     else
-        return url.pathOrURL ();
+        return url.pathOrUrl ();
 }
 
-QString kpDocument::prettyURL () const
+QString kpDocument::prettyUrl () const
 {
     return prettyURLForURL (m_url);
 }
Index: kpmainwindow.cpp
===================================================================
--- kpmainwindow.cpp	(revision 543519)
+++ kpmainwindow.cpp	(revision 543520)
@@ -879,7 +879,7 @@ void kpMainWindow::slotUpdateCaption ()
 {
     if (m_document)
     {
-        setCaption (m_configShowPath ? m_document->prettyURL ()
+        setCaption (m_configShowPath ? m_document->prettyUrl ()
                                      : m_document->prettyFilename (),
                     m_document->isModified ());
     }
Index: kpdocument.h
===================================================================
--- kpdocument.h	(revision 543519)
+++ kpdocument.h	(revision 543520)
@@ -117,7 +117,7 @@ public:
 
     // (will convert: empty URL --> "Untitled")
     static QString prettyURLForURL (const KUrl &url);
-    QString prettyURL () const;
+    QString prettyUrl () const;
 
     // (will convert: empty URL --> "Untitled")
     static QString prettyFilenameForURL (const KUrl &url);
