Index: tools/kptoolresizescale.cpp
===================================================================
--- tools/kptoolresizescale.cpp	(revision 521414)
+++ tools/kptoolresizescale.cpp	(revision 521415)
@@ -609,13 +609,13 @@ static QIcon toolButtonIconSet (const QS
 
     QPixmap disabledIcon = KGlobal::iconLoader ()->iconEffect ()->apply (
         UserIcon (iconName),
-        KIcon::Toolbar, KIcon::DisabledState);
+        K3Icon::Toolbar, K3Icon::DisabledState);
 
     const QPixmap iconSetNormalIcon = iconSet.pixmap (QIcon::Small,
                                                       QIcon::Normal);
 
     // I bet past or future versions of KIconEffect::apply() resize the
-    // disabled icon if we claim it's in group KIcon::Toolbar.  So resize
+    // disabled icon if we claim it's in group K3Icon::Toolbar.  So resize
     // it to match the QIcon::Normal icon, just in case.
     disabledIcon = kpPixmapFX::scale (disabledIcon,
                                       iconSetNormalIcon.width (),
Index: widgets/kptooltoolbar.cpp
===================================================================
--- widgets/kptooltoolbar.cpp	(revision 521414)
+++ widgets/kptooltoolbar.cpp	(revision 521415)
@@ -213,9 +213,9 @@ int kpToolToolBar::defaultIconSize ()
     #endif
 
         if (desktopSize.width () >= 1024 && desktopSize.height () >= 768)
-            m_defaultIconSize = KIcon::SizeSmallMedium/*22x22*/;
+            m_defaultIconSize = K3Icon::SizeSmallMedium/*22x22*/;
         else
-            m_defaultIconSize = KIcon::SizeSmall/*16x16*/;
+            m_defaultIconSize = K3Icon::SizeSmall/*16x16*/;
     }
 
 
