Index: kptool.cpp
===================================================================
--- kptool.cpp	(revision 528185)
+++ kptool.cpp	(revision 528186)
@@ -194,9 +194,8 @@ QString kpTool::toolTipForTextAndShortcu
         const KKeySequence seq = shortcut.seq (i);
         if (seq.count () == 1 && containsSingleKeyTrigger (seq))
         {
-            return i18n ("<Tool Name> (<Single Accel Key>)",
-                         "%1 (%2)")
-                       .arg (text, seq.toString ());
+            return i18nc ("<Tool Name> (<Single Accel Key>)",
+                          "%1 (%2)", text, seq.toString ());
         }
     }
 
@@ -1520,7 +1519,7 @@ void kpTool::setUserMessage (const QStri
     if (m_userMessage.isEmpty ())
         m_userMessage = text ();
     else
-        m_userMessage.prepend (i18n ("%1: ").arg (text ()));
+        m_userMessage.prepend (i18n ("%1: ", text ()));
 
     emit userMessageChanged (m_userMessage);
 }
Index: kpdocument.cpp
===================================================================
--- kpdocument.cpp	(revision 528185)
+++ kpdocument.cpp	(revision 528186)
@@ -147,8 +147,8 @@ QPixmap kpDocument::getPixmapFromFile (c
         if (!suppressDoesntExistDialog)
         {
             KMessageBox::sorry (parent,
-                                i18n ("Could not open \"%1\".")
-                                    .arg (kpDocument::prettyFilenameForURL (url)));
+                                i18n ("Could not open \"%1\".",
+                                      kpDocument::prettyFilenameForURL (url)));
         }
 
         return QPixmap ();
@@ -179,8 +179,8 @@ QPixmap kpDocument::getPixmapFromFile (c
     if (detectedMimeType.isEmpty ())
     {
         KMessageBox::sorry (parent,
-                            i18n ("Could not open \"%1\" - unknown mimetype.")
-                                .arg (kpDocument::prettyFilenameForURL (url)));
+                            i18n ("Could not open \"%1\" - unknown mimetype.",
+                                  kpDocument::prettyFilenameForURL (url)));
         KIO::NetAccess::removeTempFile (tempFile);
         return QPixmap ();
     }
@@ -191,8 +191,8 @@ QPixmap kpDocument::getPixmapFromFile (c
     {
         KMessageBox::sorry (parent,
                             i18n ("Could not open \"%1\" - unsupported image format.\n"
-                                  "The file may be corrupt.")
-                                .arg (kpDocument::prettyFilenameForURL (url)));
+                                  "The file may be corrupt.",
+                                  kpDocument::prettyFilenameForURL (url)));
         KIO::NetAccess::removeTempFile (tempFile);
         return QPixmap ();
     }
@@ -251,7 +251,7 @@ QPixmap kpDocument::getPixmapFromFile (c
 
     QPixmap newPixmap = kpPixmapFX::convertToPixmapAsLosslessAsPossible (image,
         kpPixmapFX::WarnAboutLossInfo (
-             i18n ("The image \"%1\""
+             ki18n ("The image \"%1\""
                    " may have more colors than the current screen mode."
                    " In order to display it, some colors may be changed."
                    " Try increasing your screen depth to at least %2bpp."
@@ -260,18 +260,18 @@ QPixmap kpDocument::getPixmapFromFile (c
 
                    " contains translucency which is not fully"
                    " supported. The translucency data will be"
-                   " approximated with a 1-bit transparency mask.")
-                 .arg (prettyFilenameForURL (url)),
-             i18n ("The image \"%1\""
+                   " approximated with a 1-bit transparency mask.",
+                   prettyFilenameForURL (url)),
+             ki18n ("The image \"%1\""
                    " may have more colors than the current screen mode."
                    " In order to display it, some colors may be changed."
-                   " Try increasing your screen depth to at least %2bpp.")
-                 .arg (prettyFilenameForURL (url)),
+                   " Try increasing your screen depth to at least %2bpp.",
+                   prettyFilenameForURL (url)),
              i18n ("The image \"%1\""
                    " contains translucency which is not fully"
                    " supported. The translucency data will be"
-                   " approximated with a 1-bit transparency mask.")
-                .arg (prettyFilenameForURL (url)),
+                   " approximated with a 1-bit transparency mask.",
+                  prettyFilenameForURL (url)),
             "docOpen",
             parent));
 
@@ -313,8 +313,8 @@ QPixmap kpDocument::getPixmapFromFile (c
     if (newPixmap.isNull ())
     {
         KMessageBox::sorry (parent,
-                            i18n ("Could not open \"%1\" - out of graphics memory.")
-                                .arg (kpDocument::prettyFilenameForURL (url)));
+                            i18n ("Could not open \"%1\" - out of graphics memory.",
+                                  kpDocument::prettyFilenameForURL (url)));
         KIO::NetAccess::removeTempFile (tempFile);
         return QPixmap ();
     }
@@ -413,9 +413,9 @@ bool kpDocument::save (bool overwritePro
         KMessageBox::detailedError (m_mainWindow,
             i18n ("Could not save image - insufficient information."),
             i18n ("URL: %1\n"
-                  "Mimetype: %2")
-                .arg (prettyURL ())
-                .arg (m_saveOptions->mimeType ().isEmpty () ?
+                  "Mimetype: %2",
+                  prettyURL (),
+                  m_saveOptions->mimeType ().isEmpty () ?
                           i18n ("<empty>") :
                           m_saveOptions->mimeType ()),
             i18n ("Internal Error"));
@@ -454,8 +454,8 @@ bool kpDocument::lossyPromptContinue (co
                 i18n ("<qt><p>The <b>%1</b> format may not be able"
                       " to preserve all of the image's color information.</p>"
 
-                      "<p>Are you sure you want to save in this format?</p></qt>")
-                    .arg (KMimeType::mimeType (saveOptions.mimeType ())->comment ()),
+                      "<p>Are you sure you want to save in this format?</p></qt>",
+                      KMimeType::mimeType (saveOptions.mimeType ())->comment ()),
                 // TODO: caption misleading for lossless formats that have
                 //       low maximum colour depth
                 i18n ("Lossy File Format"),
@@ -471,8 +471,8 @@ bool kpDocument::lossyPromptContinue (co
 
                         " Any transparency will also be removed.</p>"
 
-                        "<p>Are you sure you want to save at this color depth?</p></qt>")
-                    .arg (saveOptions.colorDepth ()),
+                        "<p>Are you sure you want to save at this color depth?</p></qt>",
+                      saveOptions.colorDepth ()),
                 i18n ("Low Color Depth"),
                 KStdGuiItem::save (),
                 QString::fromLatin1 ("SaveAtLowColorDepthDontAskAgain")));
@@ -604,8 +604,8 @@ bool kpDocument::savePixmapToFile (const
     {
         int result = KMessageBox::warningContinueCancel (parent,
             i18n ("A document called \"%1\" already exists.\n"
-                  "Do you want to overwrite it?")
-                .arg (prettyFilenameForURL (url)),
+                  "Do you want to overwrite it?",
+                  prettyFilenameForURL (url)),
             QString::null,
             i18n ("Overwrite"));
 
@@ -676,8 +676,8 @@ bool kpDocument::savePixmapToFile (const
 
         // TODO: use file.errorString()
         KMessageBox::error (parent,
-                            i18n ("Could not save as \"%1\".")
-                                .arg (kpDocument::prettyFilenameForURL (url)));
+                            i18n ("Could not save as \"%1\".",
+                                  kpDocument::prettyFilenameForURL (url)));
 
         return false;
     }
Index: tools/kptoolresizescale.cpp
===================================================================
--- tools/kptoolresizescale.cpp	(revision 528185)
+++ tools/kptoolresizescale.cpp	(revision 528186)
@@ -1102,7 +1102,8 @@ void kpToolResizeScaleDialog::accept ()
     }
 
 
-    QString message, caption, continueButtonText;
+    KLocalizedString message;
+    QString caption, continueButtonText;
 
     // Note: If eText, can't Scale nor SmoothScale.
     //       If eSelection, can't Resize.
@@ -1114,7 +1115,7 @@ void kpToolResizeScaleDialog::accept ()
         if (actionTarget == eText)
         {
             message =
-                i18n ("<qt><p>Resizing the text box to %1x%2"
+                ki18n ("<qt><p>Resizing the text box to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -1128,7 +1129,7 @@ void kpToolResizeScaleDialog::accept ()
         else if (actionTarget == eImage)
         {
             message =
-                i18n ("<qt><p>Resizing the image to %1x%2"
+                ki18n ("<qt><p>Resizing the image to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -1146,7 +1147,7 @@ void kpToolResizeScaleDialog::accept ()
         if (actionTarget == eImage)
         {
             message =
-                i18n ("<qt><p>Scaling the image to %1x%2"
+                ki18n ("<qt><p>Scaling the image to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -1160,7 +1161,7 @@ void kpToolResizeScaleDialog::accept ()
         else if (actionTarget == eSelection)
         {
             message =
-                i18n ("<qt><p>Scaling the selection to %1x%2"
+                ki18n ("<qt><p>Scaling the selection to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -1178,7 +1179,7 @@ void kpToolResizeScaleDialog::accept ()
         if (actionTarget == eImage)
         {
             message =
-                i18n ("<qt><p>Smooth Scaling the image to %1x%2"
+                ki18n ("<qt><p>Smooth Scaling the image to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -1192,7 +1193,7 @@ void kpToolResizeScaleDialog::accept ()
         else if (actionTarget == eSelection)
         {
             message =
-                i18n ("<qt><p>Smooth Scaling the selection to %1x%2"
+                ki18n ("<qt><p>Smooth Scaling the selection to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -1211,7 +1212,7 @@ void kpToolResizeScaleDialog::accept ()
     if (kpTool::warnIfBigImageSize (originalWidth (),
             originalHeight (),
             imageWidth (), imageHeight (),
-            message.arg (imageWidth ()).arg (imageHeight ()),
+            message.subs (imageWidth ()).subs (imageHeight ()).toString (),
             caption,
             continueButtonText,
             this))
Index: tools/kptoolskew.cpp
===================================================================
--- tools/kptoolskew.cpp	(revision 528185)
+++ tools/kptoolskew.cpp	(revision 528186)
@@ -82,7 +82,7 @@ QString kpToolSkewCommand::name () const
     QString opName = i18n ("Skew");
 
     if (m_actOnSelection)
-        return i18n ("Selection: %1").arg (opName);
+        return i18n ("Selection: %1", opName);
     else
         return opName;
 }
@@ -396,14 +396,15 @@ bool kpToolSkewDialog::isNoOp () const
 // private slot virtual [base QDialog]
 void kpToolSkewDialog::accept ()
 {
-    QString message, caption, continueButtonText;
+    KLocalizedString message;
+    QString caption, continueButtonText;
 
     if (document ()->selection ())
     {
         if (!document ()->selection ()->isText ())
         {
             message =
-                i18n ("<qt><p>Skewing the selection to %1x%2"
+                ki18n ("<qt><p>Skewing the selection to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -418,7 +419,7 @@ void kpToolSkewDialog::accept ()
     else
     {
         message =
-            i18n ("<qt><p>Skewing the image to %1x%2"
+            ki18n ("<qt><p>Skewing the image to %1x%2"
                   " may take a substantial amount of memory."
                   " This can reduce system"
                   " responsiveness and cause other application resource"
@@ -437,7 +438,7 @@ void kpToolSkewDialog::accept ()
     if (kpTool::warnIfBigImageSize (m_oldWidth,
             m_oldHeight,
             newWidth, newHeight,
-            message.arg (newWidth).arg (newHeight),
+            message.subs (newWidth).subs (newHeight).toString (),
             caption,
             continueButtonText,
             this))
Index: tools/kptoolconverttograyscale.cpp
===================================================================
--- tools/kptoolconverttograyscale.cpp	(revision 528185)
+++ tools/kptoolconverttograyscale.cpp	(revision 528186)
@@ -59,7 +59,7 @@ QString kpToolConvertToGrayscaleCommand:
     QString opName = i18n ("Reduce to Grayscale");
 
     if (m_actOnSelection)
-        return i18n ("Selection: %1").arg (opName);
+        return i18n ("Selection: %1", opName);
     else
         return opName;
 }
Index: tools/kptoolselection.cpp
===================================================================
--- tools/kptoolselection.cpp	(revision 528185)
+++ tools/kptoolselection.cpp	(revision 528186)
@@ -1134,8 +1134,8 @@ void kpToolSelection::endDraw (const QPo
         {
             if (m_currentMoveCommandIsSmear)
             {
-                cmd = new kpMacroCommand (i18n ("%1: Smear")
-                                            .arg (document ()->selection ()->name ()),
+                cmd = new kpMacroCommand (i18n ("%1: Smear",
+                                              document ()->selection ()->name ()),
                                         mainWindow ());
             }
             else
Index: tools/kptoolclear.cpp
===================================================================
--- tools/kptoolclear.cpp	(revision 528185)
+++ tools/kptoolclear.cpp	(revision 528186)
@@ -71,7 +71,7 @@ QString kpToolClearCommand::name () cons
     QString opName = i18n ("Clear");
 
     if (m_actOnSelection)
-        return i18n ("Selection: %1").arg (opName);
+        return i18n ("Selection: %1", opName);
     else
         return opName;
 }
Index: tools/kptoolflip.cpp
===================================================================
--- tools/kptoolflip.cpp	(revision 528185)
+++ tools/kptoolflip.cpp	(revision 528186)
@@ -88,7 +88,7 @@ QString kpToolFlipCommand::name () const
 
 
     if (m_actOnSelection)
-        return i18n ("Selection: %1").arg (opName);
+        return i18n ("Selection: %1", opName);
     else
         return opName;
 }
Index: tools/kptoolrotate.cpp
===================================================================
--- tools/kptoolrotate.cpp	(revision 528185)
+++ tools/kptoolrotate.cpp	(revision 528186)
@@ -78,7 +78,7 @@ QString kpToolRotateCommand::name () con
     QString opName = i18n ("Rotate");
 
     if (m_actOnSelection)
-        return i18n ("Selection: %1").arg (opName);
+        return i18n ("Selection: %1", opName);
     else
         return opName;
 }
@@ -450,14 +450,15 @@ void kpToolRotateDialog::slotUpdate ()
 // private slot virtual [base QDialog]
 void kpToolRotateDialog::accept ()
 {
-    QString message, caption, continueButtonText;
+    KLocalizedString message;
+    QString caption, continueButtonText;
 
     if (document ()->selection ())
     {
         if (!document ()->selection ()->isText ())
         {
             message =
-                i18n ("<qt><p>Rotating the selection to %1x%2"
+                ki18n ("<qt><p>Rotating the selection to %1x%2"
                       " may take a substantial amount of memory."
                       " This can reduce system"
                       " responsiveness and cause other application resource"
@@ -472,7 +473,7 @@ void kpToolRotateDialog::accept ()
     else
     {
         message =
-            i18n ("<qt><p>Rotating the image to %1x%2"
+            ki18n ("<qt><p>Rotating the image to %1x%2"
                   " may take a substantial amount of memory."
                   " This can reduce system"
                   " responsiveness and cause other application resource"
@@ -491,7 +492,7 @@ void kpToolRotateDialog::accept ()
     if (kpTool::warnIfBigImageSize (m_oldWidth,
             m_oldHeight,
             newWidth, newHeight,
-            message.arg (newWidth).arg (newHeight),
+            message.subs (newWidth).subs (newHeight).toString (),
             caption,
             continueButtonText,
             this))
Index: tools/kptoolpreviewdialog.cpp
===================================================================
--- tools/kptoolpreviewdialog.cpp	(revision 528185)
+++ tools/kptoolpreviewdialog.cpp	(revision 528186)
@@ -134,9 +134,9 @@ void kpToolPreviewDialog::createDimensio
     QString originalDimensions;
     if (document ())
     {
-         originalDimensions = i18n ("%1 x %2")
-                                  .arg (m_oldWidth)
-                                  .arg (m_oldHeight);
+         originalDimensions = i18n ("%1 x %2",
+                                    m_oldWidth,
+                                    m_oldHeight);
 
          // Stop the Dimensions Group Box from resizing so often
          const QString minimumLengthString ("100000 x 100000");
@@ -221,9 +221,9 @@ void kpToolPreviewDialog::updateDimensio
     kDebug () << "kpToolPreviewDialog::updateDimensions(): newDim=" << newDim << endl;
 #endif
 
-    QString newDimString = i18n ("%1 x %2")
-                               .arg (newDim.width ())
-                               .arg (newDim.height ());
+    QString newDimString = i18n ("%1 x %2",
+                                 newDim.width (),
+                                 newDim.height ());
     m_afterTransformDimensionsLabel->setText (newDimString);
 }
 
Index: kpcommandhistory.cpp
===================================================================
--- kpcommandhistory.cpp	(revision 528185)
+++ kpcommandhistory.cpp	(revision 528186)
@@ -624,7 +624,7 @@ QString kpCommandHistoryBase::undoAction
     kpCommand *undoCommand = nextUndoCommand ();
 
     if (undoCommand)
-        return i18n ("&Undo: %1").arg (undoCommand->name ());
+        return i18n ("&Undo: %1", undoCommand->name ());
     else
         return i18n ("&Undo");
 }
@@ -635,7 +635,7 @@ QString kpCommandHistoryBase::redoAction
     kpCommand *redoCommand = nextRedoCommand ();
 
     if (redoCommand)
-        return i18n ("&Redo: %1").arg (redoCommand->name ());
+        return i18n ("&Redo: %1", redoCommand->name ());
     else
         return i18n ("&Redo");
 }
@@ -783,7 +783,7 @@ static void populatePopupMenu (KMenu *po
     int i = 0;
     while (i < 10 && it != commandList.end ())
     {
-        popupMenu->insertItem (i18n ("%1: %2").arg (undoOrRedo).arg ((*it)->name ()), i/*id*/);
+        popupMenu->insertItem (i18n ("%1: %2", undoOrRedo, (*it)->name ()), i/*id*/);
         i++, it++;
     }
 
@@ -793,7 +793,7 @@ static void populatePopupMenu (KMenu *po
     {
         // TODO: maybe have a scrollview show all the items instead
         KPopupTitle *title = new KPopupTitle (popupMenu);
-        title->setTitle (i18n ("%n more item", "%n more items",
+        title->setTitle (i18np ("%n more item", "%n more items",
                                commandList.size () - i));
 
         popupMenu->insertItem (title);
Index: kpmainwindow_file.cpp
===================================================================
--- kpmainwindow_file.cpp	(revision 528185)
+++ kpmainwindow_file.cpp	(revision 528186)
@@ -706,8 +706,8 @@ bool kpMainWindow::slotReload ()
             result = KMessageBox::warningContinueCancel (this,
                          i18n ("The document \"%1\" has been modified.\n"
                                "Reloading will lose all changes since you last saved it.\n"
-                               "Are you sure?")
-                             .arg (m_document->prettyFilename ()),
+                               "Are you sure?",
+                               m_document->prettyFilename ()),
                          QString::null/*caption*/,
                          i18n ("&Reload"));
         }
@@ -716,8 +716,8 @@ bool kpMainWindow::slotReload ()
             result = KMessageBox::warningContinueCancel (this,
                          i18n ("The document \"%1\" has been modified.\n"
                                "Reloading will lose all changes.\n"
-                               "Are you sure?")
-                             .arg (m_document->prettyFilename ()),
+                               "Are you sure?",
+                               m_document->prettyFilename ()),
                          QString::null/*caption*/,
                          i18n ("&Reload"));
         }
Index: kpmainwindow.cpp
===================================================================
--- kpmainwindow.cpp	(revision 528185)
+++ kpmainwindow.cpp	(revision 528186)
@@ -685,8 +685,8 @@ bool kpMainWindow::queryClose ()
 
     int result = KMessageBox::warningYesNoCancel (this,
                      i18n ("The document \"%1\" has been modified.\n"
-                           "Do you want to save it?")
-                         .arg (m_document->prettyFilename ()),
+                           "Do you want to save it?",
+                           m_document->prettyFilename ()),
                     QString::null/*caption*/,
                     KStdGuiItem::save (), KStdGuiItem::discard ());
 
Index: kpmainwindow_edit.cpp
===================================================================
--- kpmainwindow_edit.cpp	(revision 528185)
+++ kpmainwindow_edit.cpp	(revision 528186)
@@ -65,7 +65,7 @@
 kpPixmapFX::WarnAboutLossInfo kpMainWindow::pasteWarnAboutLossInfo ()
 {
     return kpPixmapFX::WarnAboutLossInfo (
-               i18n ("The image to be pasted"
+               ki18n ("The image to be pasted"
                      " may have more colors than the current screen mode."
                      " In order to display it, some colors may be changed."
                      " Try increasing your screen depth to at least %1bpp."
@@ -75,7 +75,7 @@ kpPixmapFX::WarnAboutLossInfo kpMainWind
                      " contains translucency which is not fully"
                      " supported. The translucency data will be"
                      " approximated with a 1-bit transparency mask."),
-               i18n ("The image to be pasted"
+               ki18n ("The image to be pasted"
                      " may have more colors than the current screen mode."
                      " In order to display it, some colors may be changed."
                      " Try increasing your screen depth to at least %1bpp."),
Index: views/kpzoomedthumbnailview.cpp
===================================================================
--- views/kpzoomedthumbnailview.cpp	(revision 528185)
+++ views/kpzoomedthumbnailview.cpp	(revision 528186)
@@ -63,7 +63,7 @@ kpZoomedThumbnailView::~kpZoomedThumbnai
 // public virtual [base kpThumbnailView]
 QString kpZoomedThumbnailView::caption () const
 {
-    return i18n ("%1% - Thumbnail").arg (zoomLevelX ());
+    return i18n ("%1% - Thumbnail", zoomLevelX ());
 }
 
 
Index: widgets/kptoolwidgetspraycansize.cpp
===================================================================
--- widgets/kptoolwidgetspraycansize.cpp	(revision 528185)
+++ widgets/kptoolwidgetspraycansize.cpp	(revision 528186)
@@ -89,7 +89,7 @@ kpToolWidgetSpraycanSize::kpToolWidgetSp
 
         pixmap.setMask (mask);
         
-        addOption (pixmap, i18n ("%1x%2").arg (s).arg (s)/*tooltip*/);
+        addOption (pixmap, i18n ("%1x%2", s, s)/*tooltip*/);
         if (i == 1)
             startNewOptionRow ();
     }
Index: widgets/kptoolwidgeterasersize.cpp
===================================================================
--- widgets/kptoolwidgeterasersize.cpp	(revision 528185)
+++ widgets/kptoolwidgeterasersize.cpp	(revision 528186)
@@ -87,7 +87,7 @@ kpToolWidgetEraserSize::kpToolWidgetEras
         previewPixmap.setMask (mask);
 
 
-        addOption (previewPixmap, i18n ("%1x%2").arg (s).arg (s)/*tooltip*/);
+        addOption (previewPixmap, i18n ("%1x%2", s, s)/*tooltip*/);
 
 
         cursorPixmap++;
Index: widgets/kptoolwidgetbrush.cpp
===================================================================
--- widgets/kptoolwidgetbrush.cpp	(revision 528185)
+++ widgets/kptoolwidgetbrush.cpp	(revision 528186)
@@ -159,7 +159,7 @@ QString kpToolWidgetBrush::brushName (in
     if (shapeName.isEmpty ())
         return QString::null;
     
-    return i18n ("%1x%2 %3").arg (s).arg (s).arg (shapeName);
+    return i18n ("%1x%2 %3", s, s, shapeName);
 }
 
 QPixmap kpToolWidgetBrush::brush () const
Index: widgets/kpcolortoolbar.cpp
===================================================================
--- widgets/kpcolortoolbar.cpp	(revision 528185)
+++ widgets/kpcolortoolbar.cpp	(revision 528186)
@@ -912,7 +912,7 @@ void kpColorSimilarityToolBarItem::setCo
 
     kpColorSimilarityCube::setColorSimilarity (similarity);
     if (similarity > 0)
-        QToolTip::add (this, i18n ("Color similarity: %1%").arg (qRound (similarity * 100)));
+        QToolTip::add (this, i18n ("Color similarity: %1%", qRound (similarity * 100)));
     else
         QToolTip::add (this, i18n ("Color similarity: Exact"));
 
Index: kpmainwindow_statusbar.cpp
===================================================================
--- kpmainwindow_statusbar.cpp	(revision 528185)
+++ kpmainwindow_statusbar.cpp	(revision 528186)
@@ -142,18 +142,18 @@ void kpMainWindow::setStatusBarShapePoin
     }
     else if (endPoint == KP_INVALID_POINT)
     {
-        statusBar ()->changeItem (i18n ("%1,%2")
-                                    .arg (startPoint.x ())
-                                    .arg (startPoint.y ()),
+        statusBar ()->changeItem (i18n ("%1,%2",
+                                      startPoint.x (),
+                                      startPoint.y ()),
                                   StatusBarItemShapePoints);
     }
     else
     {
-        statusBar ()->changeItem (i18n  ("%1,%2 - %3,%4")
-                                    .arg (startPoint.x ())
-                                    .arg (startPoint.y ())
-                                    .arg (endPoint.x ())
-                                    .arg (endPoint.y ()),
+        statusBar ()->changeItem (i18n  ("%1,%2 - %3,%4",
+                                      startPoint.x (),
+                                      startPoint.y (),
+                                      endPoint.x (),
+                                      endPoint.y ()),
                                   StatusBarItemShapePoints);
     }
 
@@ -190,9 +190,9 @@ void kpMainWindow::setStatusBarShapeSize
     }
     else
     {
-        statusBar ()->changeItem (i18n ("%1x%2")
-                                    .arg (size.width ())
-                                    .arg (size.height ()),
+        statusBar ()->changeItem (i18n ("%1x%2",
+                                      size.width (),
+                                      size.height ()),
                                   StatusBarItemShapeSize);
     }
 
@@ -219,9 +219,9 @@ void kpMainWindow::setStatusBarDocSize (
     }
     else
     {
-        statusBar ()->changeItem (i18n ("%1x%2")
-                                    .arg (size.width ())
-                                    .arg (size.height ()),
+        statusBar ()->changeItem (i18n ("%1x%2",
+                                      size.width (),
+                                      size.height ()),
                                   StatusBarItemDocSize);
     }
 }
@@ -245,7 +245,7 @@ void kpMainWindow::setStatusBarDocDepth 
     }
     else
     {
-        statusBar ()->changeItem (i18n ("%1bpp").arg (depth),
+        statusBar ()->changeItem (i18n ("%1bpp", depth),
                                   StatusBarItemDocDepth);
     }
 }
@@ -269,7 +269,7 @@ void kpMainWindow::setStatusBarZoom (int
     }
     else
     {
-        statusBar ()->changeItem (i18n ("%1%").arg (zoom),
+        statusBar ()->changeItem (i18n ("%1%", zoom),
                                   StatusBarItemZoom);
     }
 }
Index: pixmapfx/kpcoloreffect.cpp
===================================================================
--- pixmapfx/kpcoloreffect.cpp	(revision 528185)
+++ pixmapfx/kpcoloreffect.cpp	(revision 528186)
@@ -60,7 +60,7 @@ kpColorEffectCommand::~kpColorEffectComm
 QString kpColorEffectCommand::name () const
 {
     if (m_actOnSelection)
-        return i18n ("Selection: %1").arg (m_name);
+        return i18n ("Selection: %1", m_name);
     else
         return m_name;
 }
Index: pixmapfx/kppixmapfx.cpp
===================================================================
--- pixmapfx/kppixmapfx.cpp	(revision 528185)
+++ pixmapfx/kppixmapfx.cpp	(revision 528186)
@@ -337,7 +337,7 @@ static void convertToPixmapWarnAboutLoss
     {
         KMessageBox::information (wali.m_parent,
             wali.m_moreColorsThanDisplayAndHasAlphaChannelMessage
-                .arg (screenDepthNeeded),
+                .subs (screenDepthNeeded).toString (),
             QString::null,  // or would you prefer "Low Screen Depth and Image Contains Transparency"? :)
             colorDepthTranslucencyDontAskAgain);
 
@@ -351,7 +351,7 @@ static void convertToPixmapWarnAboutLoss
     {
         KMessageBox::information (wali.m_parent,
             wali.m_moreColorsThanDisplayMessage
-                .arg (screenDepthNeeded),
+                .subs (screenDepthNeeded).toString (),
             i18n ("Low Screen Depth"),
             colorDepthDontAskAgain);
     }
Index: pixmapfx/kppixmapfx.h
===================================================================
--- pixmapfx/kppixmapfx.h	(revision 528185)
+++ pixmapfx/kppixmapfx.h	(revision 528186)
@@ -44,6 +44,8 @@ class QString;
 class QWidget;
 class QMatrix;
 
+class KLocalizedString;
+
 class kpColor;
 class kpSelection;
 
@@ -102,7 +104,7 @@ public:
     {
         // <moreColorsThanDisplayAndHasAlphaChannelMessage>:
         //
-        //     i18n ("The (image \"example.jpg\"|image from the clipboard)"
+        //     ki18n ("The (image \"example.jpg\"|image from the clipboard)"
         //           " may have more colors than the current screen mode."
         //           " In order to display it, some colors may be changed."
         //           " Try increasing your screen depth to at least %1bpp."
@@ -114,7 +116,7 @@ public:
         //           " approximated with a 1-bit transparency mask.")
         //
         // <moreColorsThanDisplayMessage>:
-        //     i18n ("The (image \"example.jpg\"|image from the clipboard)"
+        //     ki18n ("The (image \"example.jpg\"|image from the clipboard)"
         //           " may have more colors than the current screen mode."
         //           " In order to display it, some colors may be changed."
         //           " Try increasing your screen depth to at least %1bpp.")
@@ -133,8 +135,8 @@ public:
         //
         //     Dialog parent
         //
-        WarnAboutLossInfo (const QString &moreColorsThanDisplayAndHasAlphaChannelMessage,
-                const QString &moreColorsThanDisplayMessage,
+        WarnAboutLossInfo (const KLocalizedString &moreColorsThanDisplayAndHasAlphaChannelMessage,
+                const KLocalizedString &moreColorsThanDisplayMessage,
                 const QString &hasAlphaChannelMessage,
                 const QString &dontAskAgainPrefix,
                 QWidget *parent)
@@ -166,9 +168,9 @@ public:
         bool isValid () const { return m_isValid; }
 
 
-        QString m_moreColorsThanDisplayAndHasAlphaChannelMessage,
-                m_moreColorsThanDisplayMessage,
-                m_hasAlphaChannelMessage;
+        KLocalizedString m_moreColorsThanDisplayAndHasAlphaChannelMessage;
+        KLocalizedString m_moreColorsThanDisplayMessage;
+        QString m_hasAlphaChannelMessage;
         QString m_dontAskAgainPrefix;
         QWidget *m_parent;
         bool m_isValid;
Index: kpmainwindow_view.cpp
===================================================================
--- kpmainwindow_view.cpp	(revision 528185)
+++ kpmainwindow_view.cpp	(revision 528186)
@@ -214,7 +214,7 @@ int kpMainWindow::zoomLevelFromString (c
 // private
 QString kpMainWindow::zoomLevelToString (int zoomLevel)
 {
-    return i18n ("%1%").arg (zoomLevel);
+    return i18n ("%1%", zoomLevel);
 }
 
 // private
@@ -234,10 +234,10 @@ void kpMainWindow::zoomTo (int zoomLevel
         if (KMessageBox::warningContinueCancel (this,
             i18n ("Setting the zoom level to a value that is not a multiple of 100% "
                   "results in imprecise editing and redraw glitches.\n"
-                  "Do you really want to set to zoom level to %1%?")
-                .arg (zoomLevel),
+                  "Do you really want to set to zoom level to %1%?",
+                  zoomLevel),
             QString::null/*caption*/,
-            i18n ("Set Zoom Level to %1%").arg (zoomLevel),
+            i18n ("Set Zoom Level to %1%", zoomLevel),
             "DoNotAskAgain_ZoomLevelNotMultipleOf100") != KMessageBox::Continue)
         {
             zoomLevel = m_mainView->zoomLevelX ();
Index: kpdocumentsaveoptionswidget.cpp
===================================================================
--- kpdocumentsaveoptionswidget.cpp	(revision 528185)
+++ kpdocumentsaveoptionswidget.cpp	(revision 528186)
@@ -148,20 +148,8 @@ void kpDocumentSaveOptionsPreviewDialog:
                << endl;
 #endif
 
-    // HACK: I don't know if the percentage thing will work well and we're
-    //       really close to the message freeze so provide alt. texts to choose
-    //       from during the message freeze :)
-    const QString alternateText0 = i18n ("%1 bytes");
-    const QString alternateText1 = i18n ("%1 bytes (%2%)");
-    const QString alternateText2 = i18n ("%1 B");
-    const QString alternateText3 = i18n ("%1 B (%2%)");
-    const QString alternateText4 = i18n ("%1 B (approx. %2%)");
-    const QString alternateText5 = i18n ("%1B");
-    const QString alternateText6 = i18n ("%1B (%2%)");
-    const QString alternateText7 = i18n ("%1B (approx. %2%)");
-    m_fileSizeLabel->setText (i18n ("%1 bytes (approx. %2%)")
-                                   .arg (KGlobal::locale ()->formatLong (m_fileSize))
-                                   .arg (percent));
+    m_fileSizeLabel->setText (i18np ("1 byte (approx. %1%)", "%n bytes (approx. %1%)",
+                                     m_fileSize, percent));
 }
 
 // public slot
Index: kpmainwindow_help.cpp
===================================================================
--- kpmainwindow_help.cpp	(revision 528185)
+++ kpmainwindow_help.cpp	(revision 528186)
@@ -138,7 +138,9 @@ void kpMainWindow::slotHelpTakingScreens
 
             "<p>Alternatively, you may try the application"
             " <a href=\"run ksnapshot\">KSnapshot</a>."
-            "</p>"
+            "</p>",
+            // TODO: Totally bogus logic if kwin not running under same user as KolourPaint.
+            ::printScreenShortcutString ()
         );
     }
     else
@@ -162,12 +164,12 @@ void kpMainWindow::slotHelpTakingScreens
 
             "<p>Alternatively, you may try the application"
             " <a href=\"run ksnapshot\">KSnapshot</a>."
-            "</p>"
+            "</p>",
+            // TODO: Totally bogus logic if kwin not running under same user as KolourPaint.
+            ::printScreenShortcutString ()
         );
     }
 
-    // TODO: Totally bogus logic if kwin not running under same user as KolourPaint.
-    message = message.arg (::printScreenShortcutString ());
 
     // Add extra vertical space
     message += "<p>&nbsp;</p>";
Index: kpmainwindow_tools.cpp
===================================================================
--- kpmainwindow_tools.cpp	(revision 528185)
+++ kpmainwindow_tools.cpp	(revision 528186)
@@ -530,9 +530,9 @@ void kpMainWindow::slotEndedDocResize (c
                     " problems.</p>"
 
                     "<p>Are you sure want to resize the"
-                    " image?</p></qt>")
-                .arg (m_docResizeWidth)
-                .arg (m_docResizeHeight),
+                    " image?</p></qt>",
+                  m_docResizeWidth,
+                  m_docResizeHeight),
             i18n ("Resize Image?"),
             i18n ("R&esize Image"),
             this))
